src/test/java/org/jabref/logic/externalfiles/LinkedFileHandlerTest.java

```diff
@@ -0,0 +1,125 @@
+package org.jabref.logic.externalfiles;
+
+
+import org.jabref.gui.preferences.GuiPreferences;
+import org.jabref.logic.FilePreferences;
+
+import org.jabref.logic.xmp.XmpPreferences;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.LinkedFile;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.net.CookieHandler;
+import java.net.CookieManager;
+import java.net.CookiePolicy;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+class LinkedFileHandlerTest {
+    private Path tempFolder;
+    private Path tempFile;
+    private LinkedFile linkedFile;
+    private BibEntry entry;
+    private BibDatabaseContext databaseContext;
+    private final FilePreferences filePreferences = mock(FilePreferences.class);
+    private final GuiPreferences preferences = mock(GuiPreferences.class);
+
+    @BeforeEach
+    void setUp(@TempDir Path tempFolder) {
+        entry = new BibEntry().withCitationKey("asdf");
+        databaseContext = new BibDatabaseContext();
+
+        // Mock preferences
+        when(filePreferences.confirmDeleteLinkedFile()).thenReturn(true);
+        when(preferences.getFilePreferences()).thenReturn(filePreferences);
+        when(preferences.getXmpPreferences()).thenReturn(mock(XmpPreferences.class));
+
+        this.tempFolder = tempFolder;
+
+        // Check if there exists a system-wide cookie handler
+        if (CookieHandler.getDefault() == null) {
```

Comment:
Why do you need a cookie handler in the LinkedFilesTest?  It's not needed here, you don't do any downloads

Reply:
Not sure whether some AI was used here. - Hints on AI usage can be found at https://roe.dev/blog/using-ai-in-open-source.